client/src/components/PrivateRoute/PrivateRoute.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useContext } from "react";
+import { AuthContext } from "src/contexts/AuthContext";
+import { emptyUser } from "src/interfaces";
+import { Redirect, RouteProps } from "react-router-dom";
+import { LOGIN_PAGE } from "src/constants";
+import { AlertContext } from "src/contexts/AlertContext";
+
+interface PrivateRouteProps extends RouteProps {
+  component: any;
+}
+
+const PrivateRoute = (props: PrivateRouteProps) => {
```

Comment:
I understand this a higher order Route component to make sure "user" is there before routing to other pages, but PrivateRoute does not sound like a good name to me. 
Consider 
1. Adding comment to explain why this is called private route and a brief introduction towards this component
2. Rename it to AuthenticatedRoute or SafeRoute